### ISSUE TITLE
Type member incorrectly shown as `any`

### ISSUE THREAD

**Issue by srmagura** [TypeStrong/typedoc#1788]
## Search terms

`any`

## Expected Behavior

TypeDoc should not show the `email` member as having type `any` for the following code:

```ts
export type Validators = {
  email: (value: string) => boolean;
};
```

## Actual Behavior

![image](https://user-images.githubusercontent.com/801549/141233251-21e32219-28d6-4944-86b6-32a82c0a3474.png)

## Steps to reproduce the bug

1. Clone https://github.com/srmagura/typedoc-namespace-variable-repro.
2. `yarn install`
3. `yarn typedoc`

## Environment

-   Typedoc version: 0.22.8
-   TypeScript version: 4.4.4
-   Node.js version: 16.13.0
-   OS: Windows 11

**Comment by Gerrit0** [comment]
Wow, that's a weird one... should be an easy fix at least, looks like it's just a rendering issue.

**Comment by aqumus** [comment]
@Gerrit0 Is this issue been worked on by anyone? I could give it a try and raise a PR.
Any initial pointers or fix you might have thought of?

**Comment by Gerrit0** [comment]
Not as far as I know - type.tsx is going to be the file you want.

**Comment by matteobruni** [comment]
I've searched in the code and the `type.tsx` file is not handling function parameters here:

https://github.com/TypeStrong/typedoc/blob/3f0dbeab09ec61b5351f6766a0b13f37ea26c8c7/src/lib/output/themes/default/partials/type.tsx#L335

The `item.type` is undefined, and that's why `any` appears. But `item` if it's a function has `signatures` and each signature has `parameters` that are not rendered there.

https://github.com/TypeStrong/typedoc/blob/3f0dbeab09ec61b5351f6766a0b13f37ea26c8c7/src/lib/output/themes/default/partials/member.signature.title.tsx#L25

Here instead is where the parameters and signatures are handled.

It's not simple for a new contributor to handle this change, I managed the parameters, but I can't find the function return value
